backend/oasst_backend/utils/database_utils.py

```diff
@@ -34,28 +39,42 @@ def decorator(f):
         @wraps(f)
         def wrapped_f(self, *args, **kwargs):
             try:
-                for i in range(num_retries):
-                    try:
-                        result = f(self, *args, **kwargs)
-                        if auto_commit == CommitMode.COMMIT:
+                result = None
+                if auto_commit == CommitMode.COMMIT:
+                    retry_exhausted = True
+                    for i in range(num_retries):
+                        try:
+                            result = f(self, *args, **kwargs)
                             self.db.commit()
-                        elif auto_commit == CommitMode.FLUSH:
-                            self.db.flush()
-                        elif auto_commit == CommitMode.ROLLBACK:
+                            retry_exhausted = False
+                            break
+                        except PendingRollbackError as e:
+                            logger.info(f"{e}")
                             self.db.rollback()
-                        if isinstance(result, SQLModel):
-                            self.db.refresh(result)
-                        return result
-                    except OperationalError:
-                        logger.info(f"Retry {i+1}/{num_retries} after possible DB concurrent update conflict.")
+                        except OperationalError as e:
+                            if e.orig is not None and isinstance(
```

Comment:
the else case for this if seems to be missing, e.g. when we cannot retry we should re-raise the exception